include/clang/AST/Expr.h

```diff
@@ -2884,7 +2884,7 @@ class CastExpr : public Expr {
     // in the AST because they are part of the source program.  For other
     // cast expressions, the bounds are not included because they are
     // inferred by the compiler only when needed.
-    if (getStmtClass() == BoundsCastExprClass)
+    if (SubExprs[BOUNDS])
```

Comment:
Please undo this change.    This could incorrect behavior if this is an implicit or explicit cast expression and we've inferred the bounds.  They inferred bounds (which were computed by the compiler) should not be included in the AST children.